flexmeasures/data/models/data_sources.py

```diff
@@ -10,14 +10,89 @@
 from flask import current_app
 import hashlib
 
+from marshmallow import Schema
+
 
 if TYPE_CHECKING:
     from flexmeasures.data.models.user import User
 
 
-class DataGeneratorMixin:
+class DataGenerator:
+    __data_generator_base__: str | None = None
     _data_source: DataSource | None = None
 
+    _config: dict = None
+
+    _input_schema: Schema | None = None
+    _config_schema: Schema | None = None
+
+    def __init__(self, config: dict | None = None, **kwargs) -> None:
+        """Base class for the Schedulers, Reporters and Forecasters.
+
+        The configuration `config` stores static parameters, parameters that, if
+        changed, trigger the creation of a new DataSource.  Dynamic parameters, such as
+        the start date, can go into the `inputs`. See docstring of the method `DataGenerator.compute` for
```

Comment:
A quick search on this PR still gives 17 hits for _inputs_. Let's rename them to _input_, too.